lru/lru.go

```diff
@@ -45,8 +45,8 @@ func (c *Cache) Add(key Key, value interface{}) {
 
 // Get looks up a key's value from the cache.
 func (c *Cache) Get(key Key) (value interface{}, ok bool) {
-	c.lock.RLock()
-	defer c.lock.RUnlock()
+	c.lock.Lock()
+	defer c.lock.Unlock()
 	return c.cache.Get(key)
```

Comment:
this test reliably reproduces the npe before the fix, add this to ensure coverage of the bug:

```
func TestGetRace(t *testing.T) {
	// size to force eviction and exercise next,curr,prev list behavior
	lru := New(25)
	// set up parallel getters/writers on 2x len keys
	for key := 0; key < 50; key++ {
		go func(key int) {
			for {
				lru.Get(key)
				lru.Add(key, 1)
				lru.Get(key)
			}
		}(key)
	}
	// let them run
	time.Sleep(5 * time.Second)
}
```

Reply:
Updated